### PR TITLE
Feature#25 UI view

### LINKED ISSUE #37: Feature#25 - UI - View

### PR DESCRIPTION
**Pull request by paula0403**
#37  closes

### CODE REVIEW

File: src/app/View.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          return Render.elementFactory('img', {
+            src: `./static/images/avatars/dog.png`,
+          });
+        }
+        return Render.elementFactory('img', {
+          src: `./static/images/avatars/${animalName}.png`,
+        });
+      },
+    );
+    Render.render(
```
Comment (Enessetere) [inline src/app/View.ts]: use Render.removeAllChildren  before you render new view.

File: src/app/View.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    const avatarsElements = Object.values(AnimalNames).map(
+      (animalName) => {
+        if (animalName === AnimalNames.BIG_DOG) {
+          return Render.elementFactory('img', {
+            src: `./static/images/avatars/dog.png`,
+          });
+        }
+        return Render.elementFactory('img', {
+          src: `./static/images/avatars/${animalName}.png`,
+        });
```
Comment (Enessetere) [inline src/app/View.ts:31]: here you create additional img with incorrect source (no 'small dog.png' file in this path)

File: src/app/View.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        console.log(playersChosenAvatarPath);
+      });
+    });
+
+    const handleClick = () => {
+      const inputValue =
+        (<HTMLInputElement>inputName).value === ''
+          ? 'Carlos Santana'
+          : (<HTMLInputElement>inputName).value;
+      const newPlayer = new Player(
```
Comment (Enessetere) [inline src/app/View.ts]: Player should be created in GameController, but for now you can mark it to refactor

File: src/app/View.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    const rollADiceButton = Render.elementFactory(
+      'button',
+      {
+        className: 'button__dice button',
+      },
+      'Roll a dice',
+    );
+    backToMenuButton.addEventListener('click', this.renderMenuView);
+
+    Render.render(
```
Comment (Enessetere) [inline src/app/View.ts]: As before, use Render.removeAllChildren before you render new view

**Review by Enessetere — CHANGES_REQUESTED** [review]
Consider mentioned changes and adding class to image, so it will be smaller.

File: src/app/View.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    };
+    startGameButton.addEventListener('click', handleClick);
+  }
+
+  renderGameView(
+    playersChosenName: string,
+    playersChosenAvatarPath: string,
+  ): void {
+    const newPlayer = new Player(
+      playersChosenName,
```
Comment (vieraboschkova) [inline src/app/View.ts:75]: 64-67 can be removed.

File: src/app/View.ts (hunk 3 above)
Comment (synowa) [reply src/app/View.ts]: I can use that actually with my solution (for now).

**Review by synowa — APPROVED** [review]
As a solution that works not looks that's ok.
Consider changes requested by @Enessetere, of course.

File: src/app/View.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      startGameButton,
+      ...avatarsElements,
+    );
+    let playersChosenAvatarPath = '';
+    avatarsElements.forEach((el) => {
+      el.addEventListener('click', (e): void => {
+        playersChosenAvatarPath = (e.target as any).src;
+      });
+    });
+
```
Comment (vieraboschkova) [inline src/app/View.ts:53]: You may try to do as HTMLImageElement and and some styling after it is clicked (but also should remove styling from other items, not done below)
`      el.addEventListener('click', (e: Event): void => {`
        `const avatarClicked = e.target as HTMLImageElement;`
        `playersChosenAvatarPath = avatarClicked.src;`
        `avatarClicked.style.border = '4px solid lightgrey';`

File: src/app/View.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
+import { Render } from './utils/Render';
+import { Timer } from './Timer';
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+import { Player } from '../Player';
+
+export class View {
+  renderMenuView(): void {
+    const inputName = Render.elementFactory('input', {
```
Comment (vieraboschkova) [inline src/app/View.ts:9]: consider adding type definition after initialization `:HTMLElement`

File: src/app/View.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      placeholder: 'Carlos Santana',
+    });
+    const startGameButton = Render.elementFactory(
+      'button',
+      {
+        className: 'button__start button',
+      },
+      'Start Game',
+    );
+    const avatarsElements = Object.values(AnimalNames).map(
```
Comment (vieraboschkova) [inline src/app/View.ts:16]: consider adding type definition, for example `HTMLElement[]`

File: src/app/View.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+            src: `./static/images/avatars/dog.png`,
+          });
+        }
+
+        return Render.elementFactory('img', {
+          src: `./static/images/avatars/${animalName}.png`,
+        });
+      },
+    );
+    avatarsElements.splice(5, 1);
```
Comment (vieraboschkova) [inline src/app/View.ts:34]: Perhaps _.compact from lodash could work also?

File: src/app/View.ts (hunk 6 above)
Comment (synowa) [reply src/app/View.ts:53]: For now, as it is still a solution for Sprint I, we don't need any styling. The only thing we care about is that it should work.

File: src/app/View.ts (hunk 9 above)
Comment (synowa) [reply src/app/View.ts:34]: For now please transfer refactor like that for the time after we finish Sprint I.

**Review by synowa — APPROVED** [review]

File: src/app/View.ts (hunk 6 above)
Comment (vieraboschkova) [reply src/app/View.ts:53]: Yes, it doesn't need complicated styling for now, but there will be no confirmation that anything was clicked or chosen, if no styles change on click

File: styles/components/_view.scss [hunk 10]
```diff
@@ -0,0 +1,42 @@
+.input__name {
+  display: block;
+  width: 150px;
+  height: 40px;
+  padding: 10px;
+  margin: 20px auto 0;
+}
+.container-img {
```
Comment (vieraboschkova) [inline styles/components/_view.scss:8]: Consider container__image etc.

**Review by vieraboschkova — APPROVED** [review]

**Review by Enessetere — APPROVED** [review]
For now looks good.

**Pull request merged**